dpp/src/main/java/org/dashj/platform/dashpay/BlockchainIdentity.kt

```diff
@@ -446,16 +444,17 @@ class BlockchainIdentity {
         val signingKey = maybeDecryptKey(assetLockTransaction!!.assetLockPublicKey, keyParameter)
         var registeredOrError = false
         var count = 1
+        val coreHeight = if (assetLockTransaction!!.confidence.confidenceType == TransactionConfidence.ConfidenceType.BUILDING) {
+            assetLockTransaction!!.confidence.appearedAtChainHeight
+        } else {
+            DashSystem.get(wallet!!.params).blockChain.bestChainHeight
+            // this is not supported, how can we get the height?
+            // val txInfo = platform.client.getTransaction(assetLockTransaction!!.txId.toString())
+            // txInfo?.height ?: -1
+        }.toLong()
+
```

Comment:
This is moved out of the loop so that `coreHeight` is not updated every time a block is mined which would keep this function running forever.